File: internal/usecase/reachbench/contract/fixtures/ruby/symbols_tier2/main.rb
def entry
  control_positive
  symbol_positive
  opaque_dispatch(ENV.fetch("REACHBENCH_CONTROL", ""))
end

def control_positive; end
def control_unreachable; end

def opaque_dispatch(name)
  handler = { "opaque" => method(:control_opaque) }[name]
  handler.call if handler
end

def control_opaque; end
def control_no_coverage; end

def symbol_positive; end

entry

File: internal/usecase/reachbench/contract/fixtures/php/symbols_tier2/main.php
<?php
declare(strict_types=1);

function entry(): void
{
    controlPositive();
    symbolPositive();
    opaqueDispatch((string) getenv("REACHBENCH_CONTROL"));
}

function controlPositive(): void {}
function controlUnreachable(): void {}

function opaqueDispatch(string $name): void
{
    $handlers = ["opaque" => "controlOpaque"];
    if (isset($handlers[$name])) {
        $handlers[$name]();
    }
}

function controlOpaque(): void {}
function controlNoCoverage(): void {}

function symbolPositive(): void {}

entry();
